src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +30,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        String[] temp = address.split(",");
+        block = new Block(temp[0]);
+        street = new Street(temp[1]);
+        unit = new Unit(temp[2]);
+        postalCode = new PostalCode(temp[3]);
+        this.value = block.getBlock() + street.getStreet() + unit.getUnit() + postalCode.getPostalCode();
```

Comment:
Maybe keeping this variable seems redundant and may cause future problems like inconsistent data as you have introduced new ways of handling the address value.